vumi/transports/smpp/smpp_transport.py

```diff
@@ -178,6 +198,20 @@ def get_cached_message(self, message_id):
     def delete_cached_message(self, message_id):
         return self.redis.delete(message_key(message_id))
 
+    def cache_pdu(self, vumi_message_id, seq_no, pdu):
+        # We the PDU in wire format to avoid json encoding troubles.
+        key = pdu_key(seq_no)
+        expiry = self.config.submit_sm_expiry
+        return self.redis.setex(key, expiry, json.dumps(
+            {'vumi_message_id': vumi_message_id, 'pdu': pdu.get_hex()}))
```

Comment:
Should we move this `json.dumps(...)` into a `.to_json(..)` method on `CachedPDU`?